plugin/audit_log/audit_log.c

```diff
@@ -636,7 +636,8 @@ char *audit_log_general_record(char *buf, size_t buflen,
     endptr= endbuf;
     query= escape_string(event->general_query, event->general_query_length,
                          endptr, endbuf - endptr, &endptr, &full_outlen);
-    full_outlen+= full_outlen * my_charset_utf8mb4_general_ci.mbmaxlen;
+    full_outlen*= my_charset_utf8mb4_general_ci.mbmaxlen;
```

Comment:
Can you explain this change?

Reply:
It makes the `else` branch estimate `full_outlen` more accuratelty. After executing this code:
```
  if (query_length < (size_t) (endbuf - endptr))
  {
    uint errors;
    query_length= my_convert(endptr, query_length,
                             &my_charset_utf8mb4_general_ci,
                             event->general_query,
                             event->general_query_length,
                             event->general_charset, &errors);
    query= endptr;
    endptr+= query_length;

    full_outlen+= query_length;

    query= escape_string(query, query_length, endptr, endbuf - endptr,
                         &endptr, &full_outlen);
  }
```
```
full_outlen = length(utf-8 encoded query) + length(utf8-encoded escaped query)
```

this is what we do now in "else" branch:
```
full_length = query_length * 4 + escaped_query_length * 4
```
It was a bug which was mitigated by multiplying total estimate by 4. It actually showed up in audit_log_charset test. In that test utf8 encoded query was roughly 4 times longer than source query.